cpr/api.py

```diff
@@ -40,9 +47,52 @@ def replace_paths(prefix, recorded_paths):
     """Given a list of paths in a prefix, go in and do the text or binary replacement
     of hard-coded prefixes"""
     from cpr.replacement import update_prefix
-    if not _os.path.isfile(recorded_paths):
-        raise ValueError("File given for list of recorded paths, '{}', does not appear to exist".format(recorded_paths))
-    with open(recorded_paths) as f:
-        for entry in f.readlines():
-            placeholder, mode, path = entry.split()
+    from tqdm import tqdm
+
+    if not _os.path.isabs(prefix):
+        old_prefix = _os.path.join(_os.getcwd(), prefix)
+
+    if isinstance(recorded_paths, _string_types):
+        if not _os.path.isfile(recorded_paths):
+            raise ValueError("File given for list of recorded paths, '{}', does not appear to exist".format(recorded_paths))
+        with open(recorded_paths) as f:
+            recorded_paths = (_.split() for _ in f.readlines())
+    for (placeholder, mode, path) in tqdm(recorded_paths, desc="Updating embedded prefix", leave=False):
+        if placeholder != prefix:
             update_prefix(path, prefix, placeholder, mode)
+
+
+def rehome(prefix, old_prefix=None):
+    """Given a prefix that represents a folder that has been moved from another location,
+    attempt to "rehome" it - find instances of the old prefix baked into files, and adjust
+    those baked-in locations to match the new location."""
+    import sys
```

Comment:
Why is `sys` imported here opposed to the top of the module?

Reply:
Could be at the top as a renamed import (like os is).  It's not used in any other functions, so it can be here and will only be imported when a user uses this function.

I do those things in api modules to hide everything but exactly what I want in the namespace.  It makes it easier to explore the API as an outsider to see what stuff is there.

Reply:
👍